syntax = 'proto3';
/* Task service API */
service Taskapi {
    /* Add a new task and return its id */
    rpc addClient (ClientDetails) returns (Id);
    /* Delete a task by id */
    rpc delClient(Id) returns (Client);
    /* List all tasks */
    rpc listClients (Empty) returns (Clients);
}

message ClientDetails {
    string fName = 1;
    string lName = 2;
    int64 noOfTanks = 3;
}

message Client {
    int64 id = 1;
    string fName = 2;
    string lName = 3;
    int64 noOfTanks = 4;
}

message Id {
    int64 id = 1;
}


message Clients {
    repeated Client clients = 1;
}

message Empty {
}
